addons/account_ubl/tests/test_ubl.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from odoo import tools
+from odoo.tests import common
+
+import logging
+import base64
+
+from lxml import etree
+
+_logger = logging.getLogger(__name__)
+
+UBL_NAMESPACES = {
+    'cac': '{urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2}',
+    'cbc': '{urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2}',
+}
+
+class TestUBL(common.TransactionCase):
+    def test_ubl_invoice(self):
+        Invoice = self.env['account.invoice']
+        invoices_ids = Invoice.search([])
+        for invoice in invoices_ids:
```

Comment:
We assume that every EDI documents generated are valid according the xsd because they can be used directly by the user. Suppose a document has been generated in a wrong way by Odoo. It means that a very specific case exists that leads to a wrong document during the generation. I think this test is an efficient way to detect that.